core/api/src/app/accounts/get-wallet-from-account.ts

```diff
@@ -0,0 +1,23 @@
+import { CouldNotFindWalletFromUsernameAndCurrencyError } from "@/domain/errors"
+import { WalletsRepository } from "@/services/mongoose"
+
+export const getWalletFromAccount = async (
+  account: Account,
+  walletCurrency?: WalletCurrency,
+  value?: Username | PhoneNumber,
+): Promise<Wallet | RepositoryError> => {
```

Comment:
Application layer methods should not return `RepositoryErrors` please use `ApplicationError`